src/bufr/BufrParser/Query/File.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "File.h"
+
+#include "bufr_interface.h"
+
+#include "Query.h"
+#include "QuerySet.h"
+#include "DataProvider.h"
+
+
+namespace Ingester {
+namespace bufr {
+    File::File(const std::string &filename, bool isWmoFormat, const std::string &wmoTablePath) :
+            filename_(filename),
+            fileUnit_(nextFileUnit()),
+            fileUnitTable1_(nextFileUnit()),
+            fileUnitTable2_(nextFileUnit()),
+            isWmoFormat_(isWmoFormat),
+            wmoTablePath_(wmoTablePath)
+    {
+        open();
+    }
+
+    void File::open()
+    {
+        open_f(fileUnit_, filename_.c_str());
```

Comment:
Unfortunately doing this would mean I have to expose another Fortran function in NCEPLIB-bufr so that I could enquire about the status of a file. So I might just write an issue to address this in the future.